source_pubsub/source.go

```diff
@@ -49,20 +49,50 @@ func (s *Source) Parameters() map[string]sdk.Parameter {
 	return s.config.Parameters()
 }
 
-func (s *Source) Configure(_ context.Context, cfg map[string]string) error {
-	if err := sdk.Util.ParseConfig(cfg, &s.config); err != nil {
+func (s *Source) Configure(ctx context.Context, cfg map[string]string) error {
+	var config Config
+
+	if err := sdk.Util.ParseConfig(cfg, &config); err != nil {
 		return fmt.Errorf("failed to parse config - %s", err)
 	}
 
-	if err := s.config.Validate(); err != nil {
+	config, err := config.Validate(ctx)
+	if err != nil {
 		return fmt.Errorf("config failed to validate: %w", err)
 	}
 
+	s.config = config
+
 	return nil
 }
 
 func (s *Source) Open(ctx context.Context, sdkPos sdk.Position) error {
-	client, err := NewGRPCClient(ctx, s.config, sdkPos)
+	logger := sdk.Logger(ctx)
+	var parsedPositions position.Topics
+	var err error
+
+	logger.Debug().
+		Str("at", "source.open").
+		Str("position", base64.StdEncoding.EncodeToString(sdkPos)).
+		Str("topic", s.config.TopicName).
+		Str("topics", strings.Join(s.config.TopicNames, ",")).
+		Msg("Open Source Connector")
+
+	parsedPositions, err = position.ParseSDKPosition(sdkPos)
```

Comment:
Can you push this logic in `ParseSDKPosition` such that `position.ParseSDKPosition(sdkPos, s.config.TopicNames)`  does the conversion internally.